### DISCUSSION TITLE
Is there a way to choose the RTP port range?

### DISCUSSION THREAD

**Discussion by FurretUber** [bluenviron/mediamtx#2402] [Questions] [answered]
Hi, I'm trying to set up WebRTC streaming, using OBS Studio to send the stream to mediamtx. When I have a VPN connection to the server, it always works. When I don't and the server has the "default" settings (more on that later), it very rarely works.

After some testing, I noticed the RTP ports being used were all over the place: lowest I observed was around 2000, highest was around 62000.

The server in question host other services, including Asterisk. Asterisk has a small port range configured to use with RTP and, in the lucky case mediamtx uses a port on the same range as Asterisk uses, the streaming works.

To confirm the theory, I changed the server's firewall settings to allow all UDP ports, and then it always worked. So, my question is: is it possible to set the RTP port range?

Asterisk has multiple options regarding RTP. For example, there is [rtp.conf](https://github.com/asterisk/asterisk/blob/master/configs/samples/rtp.conf.sample), where it's possible to set the RTP port range, and then there are settings for the endpoints (for PJSIP) in [pjsip.conf](https://github.com/asterisk/asterisk/blob/master/configs/samples/pjsip.conf.sample#L378).

Is there a way to choose the RTP port range on mediamtx?

**Comment by aler9** [comment, marked as answer]
Hello, in case of WebRTC, you can force all RTP and RTCP packets to pass through a single UDP or TCP port by following instructions in the README:

https://github.com/bluenviron/mediamtx#connectivity-issues

[a comment by an automated account is not shown]
